opencog/openpsi/utilities.scm

```diff
@@ -326,13 +326,13 @@
 		result
 		; else check if entity is an evaluation or predicate or schema
 		(let ((type (cog-type entity)))
-			(if (or (equal? type 'PredicateNode)
-				   (equal? type 'GroundedPredicateNode)
+			(if (or (equal? type 'GroundedPredicateNode)
 				   (equal? type 'DefinedPredicateNode))
 				(set! result (cog-evaluate! (Evaluation entity (List)))))
-			(if (or (equal? type 'SchemaNode)
-            		(equal? type 'GroundedSchemaNode)
-            	    (equal? type 'DefinedSchemaeNode))
+			(if (equal? type 'PredicateNode)
+				(set! result (cog-tv entity)))
```

Comment:
@AmeBel What's the rationale behind having the value of a Predicate come from the node's TV, rather than from the TV of a wrapping EvaluationLink, like for Grounded and DefinedPredicateNodes? Here the Predicate has no arguments, but if it did have arguments, we would need to use EvaluationLink. My understanding of the interpretation of the TV of a PredicateNode is that it represents a kind of average value for the Predicate over all arguments in the domain over time. For a Predicate with no arguments, I guess the node TV might represent something like it's average value over time, while the EvaluationLink tv would be its current value. Seems to me it makes more sense to have the current value represented with an EvaluationLink for consistency with n-ary predicates and the grounded/defined predicate types, but I definitely might be off here. @ngeiswei might have something to say here.